genetic_algorithms/problems/traveling_salesman_problem/move_generator.py

```diff
@@ -0,0 +1,28 @@
+import random
+from itertools import combinations
+
+
+from genetic_algorithms.models.move_generator import MoveGenerator
+from genetic_algorithms.problems.traveling_salesman_problem.state import TravelingSalesmanState
+from genetic_algorithms.problems.traveling_salesman_problem.moves import SwapEdges
+from genetic_algorithms.problems.traveling_salesman_problem.models.edge import Edge
+from typing import Generator
+
+
+class TravelingSalesmanMoveGenerator(MoveGenerator):
+    def __init__(self, depot_idx: int):
+        self._depot_idx = depot_idx
+
+    def _is_depot_start(self, a: Edge):
+        return a.start != self._depot_idx
+
+    def _is_depot_end(self, b: Edge):
```

Comment:
Ja bym trochę zmienił te metody tak, żeby one wyglądały bardziej jak ograniczenia. 
Czyli coś w stylu 
```python
def _deport_start(self, move):
   ....
```

Tak samo dla `end`. 

To by było wygodne pod tym względem, że moglibyśmy później napisać metodę która przechodzi przez listę wszystkich ograniczeń i sprawdza to czy wygenerowany ruch jest dobry.

Czyli zrobić coś w stylu:
```python
def _is_move_satisfies_constraints(self, move):
    constraints = [self._deport_start, self._deport_end, ....]
    return all([constraint(move) for constraint in constraints]
```